storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -128,56 +308,121 @@ private void updateSupervisorsResources(Cluster cluster, Topologies topologies)
+                this.userMap.put(topologySubmitter, new User(topologySubmitter, userResourcePools.get(topologySubmitter)));
+            }
+            if (cluster.getUnassignedExecutors(td).size() > 0) {
+                LOG.debug("adding td: {} to pending queue", td.getName());
+                this.userMap.get(topologySubmitter).addTopologyToPendingQueue(td);
+            } else {
+                LOG.debug("adding td: {} to running queue with existing status: {}", td.getName(), cluster.getStatusMap().get(td.getId()));
+                this.userMap.get(topologySubmitter).addTopologyToRunningQueue(td);
+                if (cluster.getStatusMap().get(td.getId()) == null || cluster.getStatusMap().get(td.getId()).equals("")) {
+                    cluster.setStatus(td.getId(), "Fully Scheduled");
                 }
             }
         }
+    }
 
-        for (Map.Entry<String, Map<String, Map<WorkerSlot, Collection<ExecutorDetails>>>> entry : schedulingMap.entrySet()) {
-            if (cluster.getSupervisorById(entry.getKey()) != null) {
-                str.append("/** Node: " + cluster.getSupervisorById(entry.getKey()).getHost() + "-" + entry.getKey() + " **/\n");
-            } else {
-                str.append("/** Node: Unknown may be dead -" + entry.getKey() + " **/\n");
+    private void initialize(Topologies topologies, Cluster cluster) {
+        this.cluster = cluster;
+        this.topologies = topologies;
+        this.nodes = new RAS_Nodes(this.cluster, this.topologies);
+        initUsers(topologies, cluster);
+    }
+
+    /**
+     * Get resource guarantee configs
+     *
+     * @return
+     */
+    private Map<String, Map<String, Double>> getUserResourcePools() {
+        Object raw = this.conf.get(Config.RESOURCE_AWARE_SCHEDULER_USER_POOLS);
+        Map<String, Map<String, Double>> ret = new HashMap<String, Map<String, Double>>();
+
+        if (raw != null) {
+            for (Map.Entry<String, Map<String, Number>> UserPoolEntry : ((Map<String, Map<String, Number>>) raw).entrySet()) {
+                String user = UserPoolEntry.getKey();
+                ret.put(user, new HashMap<String, Double>());
+                for (Map.Entry<String, Number> resourceEntry : UserPoolEntry.getValue().entrySet()) {
+                    ret.get(user).put(resourceEntry.getKey(), resourceEntry.getValue().doubleValue());
+                }
             }
-            for (Map.Entry<String, Map<WorkerSlot, Collection<ExecutorDetails>>> topo_sched : schedulingMap.get(entry.getKey()).entrySet()) {
-                str.append("\t-->Topology: " + topo_sched.getKey() + "\n");
-                for (Map.Entry<WorkerSlot, Collection<ExecutorDetails>> ws : topo_sched.getValue().entrySet()) {
-                    str.append("\t\t->Slot [" + ws.getKey().getPort() + "] -> " + ws.getValue() + "\n");
+        }
+
+        Map fromFile = Utils.findAndReadConfigFile("user-resource-pools.yaml", false);
```

Comment:
Well that is the same policy for the multitenant scheduler.  We first look at the storm.yaml then look at user-resource-pools.yaml.  Values gotten from user-resource-pools.yaml will be the ultimate authority thus overwriting  the same values gotten from storm.yaml